### PR TITLE
Organize 7.0 migration guide

### PR DESCRIPTION
**Pull request by IanVS**
Issue: N/A

I was starting to get a bit overwhelmed by the 7.0 migration guide.  We've got a ton of fixes and improvements and changes in this release, and it's easy to miss important changes if you don't read the whole thing.

## What I did

I tried to move what I thought were the more important and more universal breaking changes up to the top, and created some sections for changes that only apply to users of particular builders or frameworks.  

I also moved some docs-related changes into the docs section, some deprecations out of breaking changes into deprecations, and created a section that applies mainly to addon authors.

All of this could use another pass with a critical eye, and some of it was judgement calls that I wouldn't feel strongly about if someone wants to adjust.

I also tried adding "_Has automigration_" for changes I know can be handled by an automigration.  There are probably others that we could add that I just don't know about.  I think that's a nice way to tell people they don't have to pay too close of attention to the change, since it will be handled for them.  And I added a note at the top about `npx storybook@next upgrade --prerelease`.

## How to test

Docs only

### CODE REVIEW

**Comment by IanVS** [comment]
Hi @jonniebigodes.  If you take a quick look at this and agree with the general principle, I'll try to keep it up-to-date with changes that are being made to the migration guide until you're able to give it enough of a review for an approval.  But if that'll take a while I'll hold off until you're ready.

File: MIGRATION.md
Comment (jonniebigodes) [inline MIGRATION.md:268]: @IanVS can you capitalize both Storybook and CLI, turning this into: "A number of these changes can be made automatically by the Storybook CLI."

File: MIGRATION.md
Comment (jonniebigodes) [inline MIGRATION.md:679]: @IanVS one more capitalization here it should be Storybook and Webpack if you wouldn't mind.

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@IanVS, this is amazing. Can't thank you enough for this body of work and help us improve the documentation to help our community 🙏 ! I left a couple of small items to address and this should be good to go. 

Let me know once you've addressed them and I'll gladly merge this.

Hope you have a great day.

Stay safe

**Comment by IanVS** [comment]
@jonniebigodes Thanks for the review, I've rebased onto the latest changes, and taken a pass at capitalizations of `Storybook`, `Webpack`, and `Vite` through the 7.0 sections (and maybe a bit further into some 6.5 I think).

I have to say, I appreciate how overwhelmingly positive and appreciative you are in your PR reviews.  :-D

**Review by jonniebigodes — APPROVED** [review]
@IanVS, thanks for promptly addressing the feedback, going above and beyond, and updating the references to 6.5 items. Appreciate it 🙏 ! I've checked, and all is good and going to merge this once the conflicts are resolved. Do you want me to take care of them, or can you do it?

Also, on a more personal note, thank you for those kind words. It means a lot :)

Hope you have a great day.

Stay safe

**Pull request merged**
